test/helpers/utils.go

```
@@ -699,3 +699,17 @@ func DualStackSupportBeta() bool {
 
 	return GetCurrentIntegration() == "" && supportedVersions(k8sVersion)
 }
+
+// CiliumEndpointSliceFeatureEnabled returns true only if the environment has a kubernetes version
+// equal to 1.21.
+func CiliumEndpointSliceFeatureEnabled() bool {
+	// CiliumEndpointSlice feature enabled only for tests with evvironments has kubernetes
+	// version 1.21.
+	k8sVersionGreaterThen120 := versioncheck.MustCompile(">=1.21.0")
+	k8sVersionGreaterThen122 := versioncheck.MustCompile("<1.22.0")
```

Is there a specific reason to restrict this to 1.21 specifically? We will likely move the CI over to 1.22 soon and will stop testing 1.21. At that point, this feature would stop being tested in CI.

---

No specific reason to restrict only to `1.21`.  In our internal discussion, we decided that we test only for `1.21` hence targeted for `1.21`.  If we want to do `1.21` and/or above. I'll be happy to update the code.

---

Please do, I think I'd rather switch the CI over to 1.22 and find out there's a bad interaction with this code rather than just silently lose coverage.